packages/frontend/src/hooks/usePieChartConfig.ts

```diff
@@ -1,23 +1,142 @@
-import { /* ApiQueryResults, Explore, */ PieChart } from '@lightdash/common';
-import { useMemo, useState } from 'react';
+import {
+    Dimension,
+    fieldId,
+    isField,
+    isMetric,
+    Metric,
+    PieChart,
+    TableCalculation,
+} from '@lightdash/common';
+import { useCallback, useMemo, useState } from 'react';
 
-const usePieChartConfig = (
+type PieChartConfig = {
+    validPieChartConfig: PieChart;
+    isDonut: boolean;
+    toggleDonut: () => void;
+    groupAdd: () => void;
+    groupChange: (prevValue: any, newValue: any) => void;
+    groupRemove: (dimensionId: any) => void;
+    groupFieldIds: (string | null)[];
+    metricId: string | null;
+    metricChange: (metricId: string | null) => void;
+};
+
+type PieChartConfigFn = (
     pieChartConfig: PieChart | undefined,
+    fields: {
+        dimensions: Dimension[];
+        metrics: Metric[];
+        customMetrics: Metric[];
+        tableCalculations: TableCalculation[];
+    },
+) => PieChartConfig;
+
+const usePieChartConfig: PieChartConfigFn = (
+    pieChartConfig,
+    { customMetrics, dimensions, metrics, tableCalculations },
+) =>
     // resultsData: ApiQueryResults | undefined,
     // explore: Explore | undefined,
```

Comment:
we can remove this no?

Reply:
this PR is work in progress and behind a feature flag. most of the commented lines are there for future tickets. in this case resultsData and explore is passed to make a transaction between for example from a bar chart to a pie chart